### PR TITLE
Live validator update from rpaas feedback

### PR DESCRIPTION
**Pull request by raych1**
- Isolate RP when running transform on cached models in livevalidator
- Fix type error in nullableTransformer
- Throw exception when getting error in transformLoadedSpecs at livevalidator initialization
- Improve wording of error messages

### CODE REVIEW

File: lib/liveValidation/liveValidator.ts [hunk 1]
```diff
@@ -215,7 +215,18 @@ export class LiveValidator {
...
-    this.loader.transformLoadedSpecs();
+    try {
+      this.loader.transformLoadedSpecs();
+    } catch (e) {
+      const errMsg = `Failed to transform loaded specs, detail error message:${e?.message}.ErrorStack:${e?.stack}`;
+      this.logging(
+        errMsg,
+        LiveValidatorLoggingLevels.error,
+        LiveValidatorLoggingTypes.error,
+        "Oav.liveValidator.initialize.transformLoadedSpecs"
```
Comment (jianyexi) [inline lib/liveValidation/liveValidator.ts:226]: do you need to add the doc for this error code ?

Comment (raych1) [reply lib/liveValidation/liveValidator.ts:230]: This is the function name output in the log.

File: ChangeLog.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # Change Log - oav
 
+## 08/17/2021 2.8.0
+
+- Isolate RP when running transform on cache model in live validator so that the transform won't fail on global cache level
```
Comment (ruowan) [inline ChangeLog.md:5]: Which file is for isolating RP when running transform?

**Review by ruowan — APPROVED** [review]

File: ChangeLog.md (hunk 2 above)
Comment (raych1) [reply ChangeLog.md:5]: The logic was added to each global transformer to continue other object transforms when hit error in transforming current object.

File: lib/liveValidation/operationSearcher.ts [hunk 3]
```diff
@@ -279,10 +293,12 @@ export class OperationSearcher {
       requestInfo.query
     );
     if (ret.matches.length === 0 && ret.reason === undefined) {
-      ret.reason = new LiveValidationError(
-        ErrorCodes.OperationNotFoundInCache.name,
-        `Could not find best match operation for verb "${requestInfo.requestMethod}" for api-version "${requestInfo.apiVersion}" and provider "${requestInfo.providerNamespace}" in the cache.`
-      );
+      meta = getOavErrorMeta(ErrorCodes.OperationNotFoundInCache.name as any, {
```
Comment (jianyexi) [inline lib/liveValidation/operationSearcher.ts:296]: is it a old function , why I can't find its definition ?

**Review by jianyexi — APPROVED** [review]

File: lib/liveValidation/operationSearcher.ts (hunk 3 above)
Comment (raych1) [reply lib/liveValidation/operationSearcher.ts:296]: It's in `errorDefinitions.ts`

**Pull request merged**